packages/block-editor/src/hooks/border-width.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockSupport } from '@wordpress/blocks';
+import { RangeControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useEditorFeature from '../components/use-editor-feature';
+import { BORDER_SUPPORT_KEY } from './border';
+import { cleanEmptyObject } from './utils';
+
+const MIN_BORDER_WIDTH = 0;
+const MAX_BORDER_WIDTH = 50;
+
+/**
+ * Inspector control for configuring border width property.
+ *
+ * @param  {Object} props  Block properties.
+ * @return {WPElement}     Border width edit element.
+ */
+export const BorderWidthEdit = ( props ) => {
+	const {
+		attributes: { style },
+		setAttributes,
+	} = props;
+
+	if ( useIsBorderWidthDisabled( props ) ) {
+		return null;
+	}
+
+	const onChange = ( newWidth ) => {
+		const newStyle = {
+			...style,
+			border: {
+				...style?.border,
+				width: newWidth,
+			},
+		};
+
+		setAttributes( { style: cleanEmptyObject( newStyle ) } );
+	};
+
+	return (
+		<RangeControl
+			value={ style?.border?.width }
+			label={ __( 'Border width' ) }
+			min={ MIN_BORDER_WIDTH }
+			max={ MAX_BORDER_WIDTH }
+			initialPosition={ 0 }
+			allowReset
+			onChange={ onChange }
+		/>
+	);
+};
+
+/**
+ * Determines if there is border width support.
+ *
+ * @param  {string|Object} blockType Block name or block type object.
+ * @return {boolean}                 Whether there is support.
+ */
+export const hasBorderWidthSupport = ( blockType ) => {
```

Comment:
this is the 3rd appearance of the same function. We can export it to a separate file, so it can be used among all 3 different files.

Reply:
I've refactored all the individual feature checks to use a single more generic function exported from `border.js`